backends/tofino/bf-asm/salu_inst.cpp

```diff
@@ -551,7 +551,7 @@ Instruction *AluOP::pass1(Table *tbl_, Table::Actions::Action *act) {
     if (k1 && (k1->value < Target::STATEFUL_ALU_CONST_MIN() ||
                k1->value > Target::STATEFUL_ALU_CONST_MAX())) {
         if (k1->value >= (INT64_C(1) << tbl->alu_size()) ||
-            k1->value < (INT64_C(-1) << (tbl->alu_size() - 1))) {
+            k1->value < (INT64_C(~0u) << (tbl->alu_size() - 1))) {
```

Comment:
Is this giving warnings about left-shifting a negative value?  I think the replacement is incorrect, as `~0u` is likely 0xffffffff (32 bits), while we need 0xffffffffffffffff (64 bits) here.  I'm not sure what is the correct way of getting that -- perhaps `~INT64_C(0)`?  Though that may well give the same warning.  Maybe `k->value < -(INT64_C(1) << (tbl->alu_size() - 1))` would work -- we're trying to test that k->value is in range for a `tbl->alu_size()` bit 2s complement value.